lib/widgets/pay_date_dropdown.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/material.dart';
+import '../theme/theme.dart';
+import 'string_constants.dart';
+
+class PayDateDropdown extends StatefulWidget {
+  @override
+  State<PayDateDropdown> createState() => _PayDateDropDownState();
+}
+
+class _PayDateDropDownState extends State<PayDateDropdown> {
+  DateTime? selectedDate;
+
+  Future<void> _selectDate(BuildContext context) async {
+    final picked = await showDatePicker(
+      context: context,
+      initialDate: selectedDate ?? DateTime.now(),
+      firstDate: DateTime(2000),
+      lastDate: DateTime(2101),
+    );
+    if (picked != null && picked != selectedDate) {
+      setState(() {
+        selectedDate = picked;
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    return Padding(
+      padding: EdgeInsets.all(theme.spacing.width.s4),
+      child: GestureDetector(
+        onTap: () => _selectDate(context),
+        child: Container(
+          padding: EdgeInsets.all(theme.spacing.width.s4),
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(theme.borderradius.small),
+          ),
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              Row(
+                mainAxisSize: MainAxisSize.min,
+                children: [
+                  Text(
+                    payDateText,
+                    style: theme.textStyle.caption.copyWith(
+                      color: theme.colors.contrastMedium,
+                    ),
+                  ),
+                  SizedBox(width: theme.sizing.width.s3),
+                  Icon(
+                    MinyIcons.outlineArrowUp,
+                    color: theme.colors.contrastMedium,
+                  ),
+                ],
+              ),
+              SizedBox(height: theme.spacing.width.s4),
+              Text(
+                selectedDate == null
+                    ? '${DateTime.now().day.toString().padLeft(2, '0')}/'
```

Comment:
Use Intl package, DateFormat() for formatting date

Reply:
I used this command to import Intl package `flutter pub add intl`  .

Reply:
please update pubspec.yaml and puspec.lock accordingly

Reply:
See you have to upload all those files that are related to you edits, like either you manually add the package in the pubspec.yaml or use  flutter pub add command, the point is use edit the file right ? So push the Pubspec.yaml file, but pubspec.lock file is autogenerated. and depends on system as well so dont push the pubspec.lock file!!